### ISSUE TITLE
Del treated as ^D on empty readline

### ISSUE THREAD

**Issue by lmorg** [gohxs/readline#5]
If the readline buf is empty then pressing the delete key (not backspace) will trigger the EOF interrupt (aka ^D).

I expect this will be desired behavior for some applications but it is highly annoying for the application I'm working on so I've created a patch that will preserve the default behaviour include an additional config property which disables the ^D behaviour.

PR to follow

**Comment by lmorg** [comment]
Pull request: #6

**Comment by lmorg** [comment]
Pull request #7

**Comment by hexasoftware** [comment]
Actually i think this should be treated differently, 

as I see in Ansi ^D (ascii 4) should be EOT (End of Transmission) 
while delete key is sent as ^[[3~ (mostly at least)
for example in bash the behavior is exactly like readline on ^D (deletes or if empty string ends the transmission)
seems chzyer merged that and treated ^[[3~ the same as ^D

https://github.com/gohxs/readline/blob/master/utils.go#L24
https://github.com/gohxs/readline/blob/master/utils.go#L117-L120 

The proper solution is to keep the current behaviour and add a new pseudo key for `delete` that doesn't exit while empty, this way you'll get Ctrl+D and delete working as suposed

**Comment by hexasoftware** [comment]
Check the branch https://github.com/gohxs/readline/tree/bugfix/delete_is_not_eot

**Comment by hexasoftware** [comment]
went to check if the change affected windows and figured out that delete key was not working at all also in chzyer version, corrected that aswell

**Comment by lmorg** [comment]
Yeah I stumbled across the same problem (del / EOT) myself when testing my patch further last night as well. Your approach is definitely the saner method in light of that (in fact I'm puzzled why they would ever be grouped in the first place - not that it matters now).

I'll pull in your branch later today and test that out. Thank you

On a tangential note, I've noticed the original dev has started committing PRs again. I'm not sure what your plans are for this fork now that the original author is active again but it appears he hasn't committed your improvements so I'll be keeping my projects (namely murex) pointing at your repo for now. Thank you for all your time on this :)
